src/pybind/mgr/orchestrator/_interface.py

```diff
@@ -1313,19 +1313,22 @@ def _match():
                 # TODO: can a DaemonDescription exist without a hostname?
                 raise err
 
-            if self.hostname == self.daemon_id:
-                # daemon_id == "hostname"
+            # use the bare hostname, not the FQDN.
+            host = self.hostname.split('.')[0]
```

Comment:
We are changing the hostname... again .... when we should use the bare name and when the FQDN?. 
This is a small snow ball running down hill... (well.. it starts to become bigger)
Take a look to: https://github.com/ceph/ceph/pull/35274

Reply:
> We are changing the hostname... again .... when we should use the bare name and when the FQDN?.

Yes, but we made a decision to use bare hostname for the the daemon id. Do you have another suggestion how to fix this specific tracker?

> This is a small snow ball running down hill... (well.. it starts to become bigger)
> Take a look to: #35274

maybe we build :snowman: ?!

Reply:
if we change the name of daemons, we have to cope with both types for at least ceph Q anyway.

Reply:
ping @jmolmo

Reply:
ping @jmolmo

Reply:
ping @jmolmo